logger_core/src/lib.rs

```diff
@@ -117,6 +118,25 @@ impl Level {
     }
 }
 
+/// Attempt to read a directory path from an environment variable. If the environment variable `envname` exists
+/// and contains a valid path - this function will create and return that path. In any case of failure,
+/// this method returns `None` (e.g. the environment variable exists but contains an empty path etc)
+pub fn directory_from_env(envname: &str) -> Option<String> {
+    let Ok(dirpath) = std::env::var(envname) else {
+        return None;
+    };
+
+    if dirpath.trim().is_empty() {
+        return None;
+    }
+
+    if std::fs::create_dir_all(&dirpath).is_ok() {
+        Some(dirpath)
+    } else {
+        None
+    }
+}
```

Comment:
```suggestion
pub fn directory_from_env_or_constant() -> String {
    if let Ok(dirpath) = std::env::var(ENV_GLIDE_LOG_DIR) {
        if !dirpath.trim().is_empty() && std::fs::create_dir_all(&dirpath).is_ok() {
            return dirpath;
        }
    }
    FILE_DIRECTORY.to_string()
}
```

Reply:
I actually prefer to return `Option` and not explicit `String` (this is the Rust way). This gives the caller more control of the return value. Maybe it's not needed now, but it might be handy to be able to differentiate between: the env variable exist with an empty string OR it could not be found at all

Reply:
I knew you were going to say that about the rusty way :)
My concern is that the function is not limited to the relevant env parameter.
You don't create a DIR from parameters, you pull unknown env parameters based on input, assuming its directory parameter, and you don't validate that the input is right (which actually not need to do, it is a constant, use the constant). 
The issue is with the input. I don't think it's correct to accept it, the function is specific to one action only, no real reason to open it to more options when clearly it's not fits to serve more.